internal/mesh/internal/mappers/common/workload_selector_util.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/resource"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v2beta1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+// MapSelector returns requests of the provided type given a workload
+// selector and tenancy. The type has to be name-aligned with the workload.
+func MapSelector(ctx context.Context,
+	client pbresource.ResourceServiceClient,
+	typ *pbresource.Type,
+	selector *pbcatalog.WorkloadSelector,
+	tenancy *pbresource.Tenancy) ([]controller.Request, error) {
+	var result []controller.Request
+
+	for _, prefix := range selector.Prefixes {
+		resp, err := client.List(ctx, &pbresource.ListRequest{
+			Type:       pbcatalog.WorkloadType,
+			Tenancy:    tenancy,
+			NamePrefix: prefix,
+		})
+		if err != nil {
+			return nil, err
+		}
+		if len(resp.Resources) == 0 {
+			return nil, fmt.Errorf("no workloads found")
```

Comment:
Should this error? What if there are several selectors and only some of them have like deleted pods at the moment?

Reply:
Great catch. We don't need to error here anymore because once a workload appears for this selector, we should be able to pick it up from the tracker.

Reply:
Did you intend to remove the error?

Reply:
Yes, thank you. It sneakily got back in after my rebases and investigations into failing tests.